### PR TITLE
【Hackathon 4 No.16】为 Paddle 新增 PoissonNLLLoss API

### PR DESCRIPTION
**Pull request by LyndonKong**
Add PoissonNLLLoss API rfc file

### CODE REVIEW

File: rfcs/APIs/20230301_api_design_for_poissonnllloss.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+## 1、相关背景
+
+paddle.nn.PoissonNLLLoss 和 paddle.nn.functional.Poisson_nll_loss API 用于泊松分布的负对数似然函数的计算。
+该函数计算公式为：
+
+$$
+\text{loss}(\text{input}, \text{target}) = \text{input} - \text{target} * \log(\text{input}) + \log(\text{target!})
+$$
+
+损失函数中的最后一项可以使用Stirling公式近似，当target的值超过1时考虑此项近似，当target的值小于等于1的时候设置为0。
```
[the remark this answers is not in the document]
Comment (LyndonKong) [reply rfcs/APIs/20230301_api_design_for_poissonnllloss.md:23]: 这里是和每一个元素都是1的Tensor进行elemen-wise的比较，小于等于1的地方mask掉对应的逼近项。在PyTorch的实现中调用的是Tensor.masked_fill(mask, value)。

File: rfcs/APIs/20230301_api_design_for_poissonnllloss.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        }
+        return unreduced;
+    }
+```
+
+无其它相关库支持该 Loss 函数。
+
+# 四、对比分析
+
+设计方案将参考PyTorch的实现，在当前版本的API当中仅实现Python版本，对于PyTorch实现中部分抛弃的参数不再做兼容。
```
[the remark this answers is not in the document]
Comment (LyndonKong) [reply rfcs/APIs/20230301_api_design_for_poissonnllloss.md]: 更新在了第四部分中

File: rfcs/APIs/20230301_api_design_for_poissonnllloss.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+| 版本号                                                       | V1.0                                           |
+| 依赖飞桨版本<input type="checkbox" class="rowselector hidden"> | Develop                                        |
+| 文件名                                                       | 20230301_api_design_for_poissonllloss.md<br> |
+
+
+# 一、概述
+
+## 1、相关背景
+
+paddle.nn.PoissonNLLLoss 和 paddle.nn.functional.Poisson_nll_loss API 用于泊松分布的负对数似然函数的计算。
```
[the remark this answers is not in the document]
Comment (LyndonKong) [reply rfcs/APIs/20230301_api_design_for_poissonnllloss.md]: 更新了对于API的描述

File: rfcs/APIs/20230301_api_design_for_poissonnllloss.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (GGBond8488) [reply rfcs/APIs/20230301_api_design_for_poissonnllloss.md]: 这个说明补充在设计文档中

[the remark this answers is not in the document]
Comment (LyndonKong) [reply rfcs/APIs/20230301_api_design_for_poissonnllloss.md]: 在设计文档和中文文档中补充了这个说明。

**Review by GGBond8488 — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
